lib/driver.js

```diff
@@ -171,9 +171,11 @@ class EspressoDriver extends BaseDriver {
       return [sessionId, caps];
     } catch (e) {
       await this.deleteSession();
-      e.message += '. If the cause might be espresso driver or server, ' +
-        'please read troubleshooting https://github.com/appium/appium-espresso-driver#contributing';
-      throw new errors.SessionNotCreatedError(e);
+      e.message += '. Check https://github.com/appium/appium-espresso-driver#troubleshooting regarding advanced session startup troubleshooting.';
+      if (e instanceof errors.SessionNotCreatedError) {
+        throw e;
+      }
+      throw new errors.SessionNotCreatedError(e.message);
```

Comment:
Please replace the `stack` property of the newly created error, otherwise we won't be able to track it properly. Or log the stack before throwing it (`log.debug(e.stack)`)